src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java

```diff
@@ -80,14 +80,21 @@ public boolean continueExecuting()
                         final World world = citizen.world;
                         IBlockState state = world.getBlockState(pos);
                         state = state.getBlock().getActualState(state, world, pos);
-                        if (state.getBlock() instanceof BlockBed && !state.getValue(BlockBed.OCCUPIED))
+                        if (state.getBlock() instanceof BlockBed
+                                && !state.getValue(BlockBed.OCCUPIED)
+                                && state.getValue(BlockBed.PART).equals(BlockBed.EnumPartType.HEAD))
                         {
                             usedBed = pos;
-                            citizen.world.notifyBlockUpdate(pos, state, state.withProperty(BlockBed.OCCUPIED, true), 0);
+                            citizen.world.setBlockState(pos, state.withProperty(BlockBed.OCCUPIED, true), 0x03);
                             return true;
                         }
                     }
                 }
+
+                if(usedBed == null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)